src/pages/home/ReportScreen.tsx

```diff
@@ -489,12 +522,46 @@ function ReportScreen({
                 !isLoading &&
                 !userLeavingStatus) ||
             shouldHideReport ||
-            (!!reportIDFromParams && !ReportUtils.isValidReportIDFromPath(reportIDFromParams)),
-        [report, reportMetadata, isLoading, shouldHideReport, isOptimisticDelete, userLeavingStatus, reportIDFromParams],
+            (!!reportIDFromRoute && !ReportUtils.isValidReportIDFromPath(reportIDFromRoute)),
+        [report, reportMetadata, isLoading, shouldHideReport, isOptimisticDelete, userLeavingStatus, reportIDFromRoute],
     );
 
     const actionListValue = useMemo((): ActionListContextType => ({flatListRef, scrollPosition, setScrollPosition}), [flatListRef, scrollPosition, setScrollPosition]);
 
+    // This helps in tracking from the moment 'route' triggers useMemo until isLoadingInitialReportActions becomes true. It prevents blinking when loading reportActions from cache.
```

Comment:
hey @perunt, I'm realizing that I'm actually quite confused by this `useEffect` and why we're changing `isLinkingToMessage` to false after `reportMetadata.isLoadingInitialReportActions` changes.

> from the moment 'route' triggers useMemo

Which `useMemo` are you referring to here?

> until isLoadingInitialReportActions becomes true

Should we add a condition in this `useEffect` such that `isLinkingToMessage` is only `true` if `reportMetadata.isLoadingInitialReportActions` is true?

```ts
useEffect(() => {
    if (!reportMetadata?.isLoadingInitialReportActions) {
        return;
    }
    InteractionManager.runAfterInteractions(() => {
        setLinkingToMessage(false);
    });
}, [reportMetadata?.isLoadingInitialReportActions]);
```

Taking this a step further, is it necessary to have this value in state, or could a live ref be a suitable replacement for the `useState`, `useLayoutEffect`, `useEffect` combo?

```ts
/**
 * Creates a mutable reference to a value, useful when you need to
 * maintain a reference to a value that may change over time without triggering re-renders.
 */
function useLiveRef<T>(value: T) {
    const ref = useRef<T>(value);
    ref.current = value;
    return ref;
}

const isLinkingToMessage = useLiveRef(!!reportActionIDFromRoute && !reportMetadata?.isLoadingInitialReportActions);
```